Update gather-logs script

jeremiahsnapp opened the pull request:
Multiple updates to the gather-logs script

This includes many improvements and fixes
from support engineers.

ChangeLog-Entry: New gather-log script gathers a lot more debugging information.

stevendanna commented:
@jeremiahsnapp Am I correct in assuming that these changes represent the newest version of the script that support has been using and testing on a daily basis with customers?

If so, :+1:.

jeremiahsnapp commented:
@stevendanna Yes most of these changes are what we've already been manually sending to customers. There are only three relatively new features.

1. using rpm/dpkg to gather versions of Chef products
2. gathering the tail of `rabbit@localhost.log` from Chef Server and Analytics Server
3. gathering rabbitmq list_queues stats from Chef Server and Analytics Server

IMO, these new features have been sufficiently tested.

It would be great if this PR could be merged before a new release of Chef Server.

Thanks!

stevendanna commented:
@jeremiahsnapp Would you mind updating the commit message to include the new features.  perhaps something like:

```
Multiple updates to the gather-logs script

This includes various improvements and fixes
from support engineers, including:

- using rpm/dpkg to gather versions of Chef Products
- gathering the tail of RabbitMQs logs
- gathering RabbitMQ queues stats

ChangeLog-Entry: New gather-log script with additional rabbitmq debugging 
information.
``

marcparadise commented on omnibus/files/private-chef-scripts/gather-logs:
```diff
@@ -55,12 +75,15 @@ tmpdir="$(mktemp -d)/$hostname/$timestamp"
     /etc/$path/$config_name \
-    /etc/$path/chef-server-running.json \
+    /etc/${path}*/*-running.json \
+    /var/opt/$path/rabbitmq/log/rabbit@localhost.log \
```
Wasn't there a relatively recent PR merged to change the name of the rabbit node? Would this also cause the log file name to change?

stevendanna replied:
Good catch! There was and I think you are correct about the effect on the log file.

I'd still be :+1: on merging this since even in that case it just means we collect a little less data  and the user will see the tail failure.  

I think the right place to feed that kind of information into this script is via:

https://github.com/chef/chef-server/blob/master/omnibus/files/private-chef-cookbooks/private-chef/templates/default/private-chef-sh.erb

which we source at the outset.

stevendanna commented:
@jeremiahsnapp I've added a RABBITMQ_NODENAME variable to the shell script we include and changed the gather-logs script to use it instead of referring directly to rabbitmq@localhost.

This should address the concern that @marcparadise had.

Other than the commit message update I mentioned, I don't see any other blockers to merging this.

chef-supermarket commented:
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @glasschef

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

chef-supermarket commented:
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

jeremiahsnapp commented:
I reworked this PR so now it's made up of smaller, more descriptive commits.

@mp @stevendanna thanks for the heads up about rabbit name change. I'm using a glob now to grab those logs.

I added an entry to the CHANGELOG.md too.
